sdk/tables/Azure.Data.Tables/src/DynamicTableEntity.cs

```diff
@@ -52,10 +52,10 @@ public string RowKey
         /// Gets or sets the entity's ETag. Set this value to '*' in order to force an overwrite to an entity as part of an update operation.
         /// </summary>
         /// <value>A string containing the ETag value for the entity.</value>
-        public string ETag
+        public ETag ETag
         {
-            get { return GetString(TableConstants.PropertyNames.Etag); }
-            set { _properties[TableConstants.PropertyNames.Etag] = value; }
+            get { return new ETag(GetString(TableConstants.PropertyNames.EtagOdata)); }
```

Comment:
Would it be better to store `ETag` as an ETag object instead of creating a new ETag each time? Deserialization would then convert the string from the service once, and vice-versa. If a user potentially uses the indexer [] to call the etag for some reason, it would return as a string. I'm not sure if that would be confusing since the entity would technically store a type other than the accepted types for Tables.